packages/blade/src/components/FileUpload/_decisions/decisions.md

```diff
@@ -0,0 +1,359 @@
+
+## Anatomy
+
+<img width="100%" src="./file-upload-anatomy.png" alt="Single File Upload">
+<img width="100%" src="./file-upload-item-anatomy.png" alt="Multiple Files Upload">
+
+## `FileUpload` Props
+
+```ts
+// Check the File type from MDN for more details: https://developer.mozilla.org/en-US/docs/Web/API/File
+interface BladeFile extends File {
+  /**
+   * The unique identifier of the file.
+   */
+  id: string;
+  /**
+   * The file's upload status.
+   */
+  status?: 'uploading' | 'success' | 'error';
+  /**
+   * The percentage of file upload completion.
+   */
+  percent?: number;
+  /**
+   * Text indicating an error state
+   */
+  errorText?: string;
+}
+
+type BladeFileList = BladeFile[];
+
+type FileUploadProps = {
+  /**
+   * Label for the FileUpload component
+   */
+  label?: string;
+  /**
+   * Accessibility label for the FileUpload component, required if label is not provided
+   */
+  accessibilityLabel?: string;
+  /**
+   * Position of the label relative to the file upload area. Desktop only prop. Default value on mobile will be 'top'
+   */
+  labelPosition?: 'top' | 'left';
+  /**
+   * Defines the selection behavior of the FileUpload component
+   */
+  selectionType?: 'single' | 'multiple';
+  /**
+   * File types that can be accepted. See [input's accept attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file#accept)
+   *
+   * Usage: accept=".jpg, .png, .pdf", accept="image/*", accept="image/png, image/jpeg, application/pdf"
+   */
+  accept?: string;
+  /**
+   * Disables or enables the FileUpload component
+   */
+  isDisabled?: boolean;
+  /**
+   * The name of the file upload input, [useful in form submissions](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#name)
+   */
+  name?: string;
+  /**
+   * Default list of files that have been uploaded, useful when the component is uncontrolled
+   */
+  defaultFileList?: BladeFileList;
+  /**
+   * List of files that have been selected/uploaded, useful when the component is controlled
+   */
+  fileList?: BladeFileList;
+  /**
+   * Limit the number of files that can be uploaded
+   */
+  maxCount?: number;
+  /**
+   * Limit the size of the uploaded files (in bytes)
+   */
+  maxSize?: number;
+  /**
+   * Callback function triggered when files are selected
+   */
+  onChange?: ({ name, fileList }: { name: string; fileList: BladeFileList }) => void;
+  /**
+   * Callback function triggered when the preview icon is clicked
+   */
+  onPreview?: (previewedFile: BladeFile) => void;
+  /**
+   * Callback function triggered when a file is removed
+   */
+  onRemove?: (removedFile: BladeFile) => void;
+  /**
+   * Callback function executed when files are dropped into the upload area
+   */
+  onDrop?: (event: React.DragEvent) => void;
+  /**
+   * State indicating whether there is an error in the FileUpload component
+   */
+  validationState?: 'none' | 'error';
```

Comment:
do we not have success state like we have in other inputs?

Reply:
No, the success state is indicated by the attached FileItem. Because if it's successful items would have been successfully attached / uploaded.

<img width="567" alt="Screenshot 2024-02-13 at 4 36 27 PM" src="https://github.com/razorpay/blade/assets/46647141/a66f1a00-20ec-4179-a0f7-d2e2582fe156">